### ISSUE TITLE
Provide "Ripple delete" feature with hotkeys

### ISSUE THREAD

**Issue by raindropsfromsky** [OpenShot/openshot-qt#5456]
When we delete a clip with BKSP, the remaining clips on the right do not shift.
Similarly, when we split the clip with A (=CTRL+J ) and D (=CTRL+L) keys, the remaining clips do not shift to the left to remove the gap.

It would be great if ripple delete function is provided through a modifier key.

For example, pressing ALT+A or SHFT+A should ripple delete instead of normal delete.
(Select the modifier key that allows these combinations.)

**Comment by leodf1** [comment]
Pretty much dev has to look at any mainstream Editor and implement the standard basic editing functions and shortcuts. I recommend Davinci Resolve.

Worry about cloud API's later.

**Comment by Colorjet3** [comment]
I am not sure I agree with this approach.  OpenShot is not Davinci Resolve, it does not pretend to be Davinci Resolve, and the developer never intended OpenShot to be a very advanced video editor, like Davinci Resolve is.

An advanced video editor, instructor, professional, should not be coming back to OpenShot and expect it to work like Davinci Resolve or be developed to become like Davinci Resolve.  This is not the goal of this project per the lead developer and it was never meant to be.

So, I ask respectfully, that enhancement requests for OpenShot should be targeted towards making things better with what is currently in Openshot.  Of course this should include bug fixes as well.

**Comment by raindropsfromsky** [comment]
@Colorjet3 Unfortunately, the comment by @leodf1 gives a color to my suggestion.
I do not believe in his approach.

I believe that Openshot has its own distinct design philosopy, which I love.

In fact, I have tried all other editors (Resolve, kdenlive, shotcut, Olive), and finally selected Openshot.
If it started copying every other mainstream editor, I'd be disappointed indeed!

All my suggestions are to increase the productivity of an Openshot user.
Please evaluate them in that light only!

**Comment by leodf1** [comment]
@Colorjet3  My comment expressly stated "_standard _basic"__ editing functions.

In all my posts, I specifically avoided any sophisticated or frivulous requests. I never asked for extra context and mouse menus for already existing actions., exporting hotkey lists or where the mousewheel scrolls.

The most simple _essential_ shortcuts, such as moving the playhead, or switching between panels are not present in OpenShot.  This is a serious hindrance to using _any_ Editor properly.

I know Openshot does not pretend to compare with the best Editors in the world, but it should at least be efficiently usable, don't you agree?  This is not my claim. It is OpenShot's claim.  From your manual:

Manual pg.17 'Keyboard Shortcuts'
"One of the easiest ways to tell the difference between a professional video editor and a novice is to simply look at how much they use the keyboard. Editors that have been in the business for some time know that a few seconds saved add up over the length of the project...'

I can personally attest that a project took me twice the time because of **two** missing shortcuts. Jumping from Property panel to Timeline and skipping back/forward with the playhead.

These two alone would deliver orders of magnitude of efficiency and usability. And it could be implemented within minutes. 

Maintaining simplicity does not mean neglecting the basics.  All my requests have been for simple basic shortcuts only. The 'Copy; shortcut wasn't even working.

I mentioned Resolve as an easy quick reference for the dev to check up on all the **basic** functions and associated keyboard shortcuts.  If the dev prefers to re-invent the wheel and spend his time coming up with his own key binds, that's fine too.

"....enhancement requests for OpenShot should be targeted towards making things better with what is currently in Openshot."  -  Yes I agree completely. That's why I never expected a Cloud API or a UI overhaul.

**Comment by raindropsfromsky** [comment]
I agree with the approach @leodf1 outlined☝️: 
Achieve more efficiency _within_ Openshot's design paradigm.
By all means, copy features from other editors, but only if they meet this criterion.
-x-x-x-

@Colorjet3 Coming back to my suggestion in this post, the idea was to introduce ripple delete, so that we do not have to move all the other clips to eliminate the gap left behind.

This proposed feature is very much aligned with Openshot's design philosophy.

Therefore, please reconsider this feature!

**Comment by Colorjet3** [comment]
Just for the record, I am in no position to approve/disapprove any enhancement requests.  I am just a volunteer (not a developer) and consider myself a novice video editor.  

I am merely having conversations and providing my opinion.  I (this project) really appreciates all of your input.

I just checked the Enhancements/Issues queue and found a couple of similar entries, in Trello.  I thought I was able to share each entry by providing a link but that doesn't seem to be working.

Please try this and let me know the result:
1. Go to https://trello.com/c/tyVbvi59/112-eliminating-blank-empty-spaces-in-between-clips
2. Are you getting "About this board" message with the option to click on "Okay, got it"?
3. If yes, click on the "Okay, got it".
4. Right below the "Log in" find the "Filters" option.
5. For keyword type "Eliminating blank" excluding the quotes.
6. This should bring up the Backlog workspace and there you should find the entry from Step 1 (not sure why the direct link isn't working, yet).
7. Now click on that card to read the details.
8. Sometimes, you may find additional references to other "Trello attachments" of related topics.

I need to figure out why the direct link isn't working.

**Comment by leodf1** [comment]
Yes.

I have been in contact with dev and he is aware of the needed functions/shortcuts. At least the most essential ones I presented him and you so far.

He is currently occupied with big major updates to the program and has to see those through.

@Colorjet3 I realize you don't owe me your time nor attention, which is why I value both.

Once again, thank you @Colorjet3 for your work here.

**Comment by raindropsfromsky** [comment]
@Colorjet3 

Yes, we have discussed those feature requests in my other thread.

But they are about deleting blanks in a track. Here, the suggestion is about using a modifier key to convert a "normal" delete operation into a ripple delete. 

If we ripple delete, we don't have to use the other command later.

**Comment by Colorjet3** [comment]
I understand this functionality better now.  It really is all related to inserting clips and moving/deleting clips.

I will need to comb through the Trello cards and update the language and the card headings to accurately reflect what the enhancement request is.

It will then be up to the lead dev to review.  If approved then he will decide how to implement.

**Comment by raindropsfromsky** [comment]
I just found that Shotcut uses a different hotkey (x) to ripple delete the selected clip(s).
No need to use a modifier key.

**Comment by JamesRichards22** [comment]
@Colorjet3 I believe with 3.3.0 this is now resolved.
